src/menu/menu_functions.rs

```diff
@@ -463,7 +505,101 @@ mod tests {
             .collect();
         let res = find_common_string(&input);
 
-        assert!(matches!(res, (Some(elem), Some(2)) if elem == &input[0]));
+        match res.0 {
+            Some(_) => assert!(false, "There shouldn't be any elements"),
+            None => assert!(res.1 == Some(0)),
+        }
+
+        // Test against haystack of size 1
+        let input: Vec<_> = source
+            .into_iter()
+            .take(1)
+            .map(|s: &str| Suggestion {
+                value: s.into(),
+                description: None,
+                extra: None,
+                span: Span::new(0, s.len()),
+                append_whitespace: false,
+            })
+            .collect();
+        let res = find_common_string(&input);
+
+        match res {
+            (Some(first), Some(index)) => assert!(
+                first.value == input.first().unwrap().value && index == first.value.len()),
+            _ => assert!(false, "There should be a result with both options as some with one element to check against"),
+        }
```

Comment:
This whole pattern is repeated, can we factor it out so the tests are more readable and we are more confident what the differences between the tests are (and avoid introducing a bug in the test fixture)